src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -34,35 +36,60 @@
 final class RemoteLanguagePackLoader implements LanguagePackLoaderInterface
 {
     /**
-     * The link from which available languages are retrieved.
+     * The languages repository base path
+     *
+     * @TODO : duplicated from {prestashop_repository}/app/config/config.yml;
+     *         used for install context or other cases where no Symfony container is available
      */
-    public const PACK_LINK = 'http://i18n.prestashop-project.org/translations/%ps_version%/available_languages.json';
+    private const LANG_REPOSITORY_BASE_PATH = 'http://i18n.prestashop-project.org';
 
     /**
-     * @var Version
+     * @var string Prestashop version
      */
     private $version;
 
     /**
-     * @param Version $version
+     * @var string represents the language repository base path
+     */
+    private $langRepositoryBasePath;
+
+    /**
+     * @param string $version Prestashop version
+     * @param string $langRepositoryBasePath should become mandatory in future release
      */
-    public function __construct(Version $version)
+    public function __construct(string $version, string $langRepositoryBasePath = self::LANG_REPOSITORY_BASE_PATH)
     {
-        $this->version = $version;
+        // Ensure version format
+        $this->version = Version::buildFromString($version)->getSemVersion();
+        $this->langRepositoryBasePath = $langRepositoryBasePath;
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return string requested URL
+     */
+    public function getLanguagePackUrl(string $locale = null): string
+    {
+        return "{$this->langRepositoryBasePath}/translations/{$this->version}/{$locale}/{$locale}.zip";
+    }
+
+    /**
+     * @return string requested URL
+     */
+    public function getLanguagePackListUrl(): string
+    {
+        return "{$this->langRepositoryBasePath}/translations/{$this->version}/available_languages.json";
     }
 
     /**
      * {@inheritdoc}
      */
     public function getLanguagePackList()
     {
-        $normalizedLink = str_replace('%ps_version%', $this->version->getSemVersion(), self::PACK_LINK);
+        $normalizedLink = $this->getLanguagePackListUrl();
         $jsonResponse = file_get_contents($normalizedLink);
-
-        $result = [];
-        if ($jsonResponse) {
-            $result = json_decode($jsonResponse, true);
-        }
+        $result = json_decode($jsonResponse, true) ?? [];
```

Comment:
I think the previous code that checked the result before parsing JSON was a good idea no?

Reply:
After discussing with @matks , I realized that i probably misunderstood your point @jolelievre ^^'
If it is about some warnings or specific returns, my code is less straight. About output, in fact it depends of the input type.Some times it is better, but not always.

Maybe you have something in your head I didn't get ? :thinking: